docs/layers/eks/design-decisions/decide-on-default-storage-class.mdx

```diff
@@ -0,0 +1,73 @@
+
+## Amazon EBS
+
+**Amazon EBS** provides high-performance block storage volumes for use with Amazon EC2 instances, suitable for a wide range of workloads.
+
+### Pros:
+- **Higher Performance:** Offers high IOPS and low latency, making it ideal for performance-critical applications.
+- **Cost-Effective:** Potentially lower costs for specific storage types and usage scenarios.
+- **Native EKS Integration:** Well-integrated with Kubernetes through the EBS CSI (Container Storage Interface) driver, facilitating seamless provisioning and management.
+- **Supports Snapshot and Backup:** Supports snapshotting for data backup, recovery, and cloning.
+
+### Cons:
+- **Single-Attach Limitation:** EBS volumes may only be attached to a single node at a time, limiting shared access across multiple pods.
+- **Data Sharing:** Data cannot be easily shared across multiple instances, requiring additional configurations or solutions for shared access.
+- **Availability Zones:** EBS volumes are confined to a single Availability Zone, limiting high availability and disaster recovery across zones. This limitation can be mitigated by configuring a `StatefulSet` with replicas spread across multiple AZs. However, for workloads using EBS-backed Persistent Volume Claims (PVCs), failover to a different AZ requires manual intervention, including provisioning a new volume in the target zone, as EBS volumes cannot be moved between zones.
+- **Non-Elastic Storage:** EBS volumes can be manually resized, but this process is not fully automated in EKS. After resizing an EBS volume, additional manual steps are required to expand the associated Persistent Volume (PV) and Persistent Volume Claim (PVC). This introduces operational complexity, especially for workloads with dynamic storage needs, as EBS lacks automatic scaling like EFS.
+
+## Recommendation
+
+Use **Amazon EBS** as the primary storage option when:
+
+	•	High performance, low-latency storage is required for workloads confined to a single Availability Zone.
+	•	The workload doesn’t require shared access across multiple Pods.
+	•	You need cost-effective storage with support for snapshots and backups.
+	•	Manual resizing of volumes is acceptable for capacity management, recognizing that failover across AZs requires manual intervention and provisioning.
+
+Consider **Amazon EFS** when:
+
+	•	Multiple Pods need concurrent read/write access to shared data across nodes.
+	•	Workloads must persist data across multiple Availability Zones for high availability, and the application does not support native replication.
+	•	Elastic, automatically scaling storage is necessary to avoid manual provisioning, especially for workloads with unpredictable growth.
+	•	You are willing to trade off higher costs and lower performance for multi-AZ durability and easier management of shared storage.
```

Comment:
```suggestion
	*	High performance, low-latency storage is required for workloads confined to a single Availability Zone.
	*	The workload doesn’t require shared access across multiple Pods.
	*	You need cost-effective storage with support for snapshots and backups.
	*	Manual resizing of volumes is acceptable for capacity management, recognizing that failover across AZs requires manual intervention and provisioning.

Consider **Amazon EFS** when:

	*	Multiple Pods need concurrent read/write access to shared data across nodes.
	*	Workloads must persist data across multiple Availability Zones for high availability, and the application does not support native replication.
	*	Elastic, automatically scaling storage is necessary to avoid manual provisioning, especially for workloads with unpredictable growth.
	*	You are willing to trade off higher costs and lower performance for multi-AZ durability and easier management of shared storage.
```